File: KisaragiTech.Dape/Config/DatabaseConfig.cs
using System.Text.Json.Serialization;

namespace KisaragiTech.Dape.Config;

public class DatabaseConfig
{
    [JsonConstructor]
    public DatabaseConfig(string host, ushort port, string user, string password)
    {
        this.Host = host;
        this.Port = port;
        this.User = user;
        this.Password = password;
    }

    [JsonPropertyName("host")]
    public string Host { get; set; }

    [JsonPropertyName("port")]
    public ushort Port { get; set; }

    [JsonPropertyName("user")]
    public string User { get; set; }

    [JsonPropertyName("password")]
    public string Password { get; set; }
}

File: src/main/scala/io/github/ksrgtech/dape/config/DatabaseConfig.scala
package io.github.ksrgtech.dape.config

import io.circe.Decoder
import io.circe.generic.semiauto.deriveDecoder

final case class DatabaseConfig(
    host: String,
    port: Int,
    user: String,
    password: String
)

object DatabaseConfig {
  implicit val decoder: Decoder[DatabaseConfig] = deriveDecoder[DatabaseConfig]
}
